server/book/orderpq.go

```diff
@@ -434,11 +441,27 @@ func (pq *OrderPQ) RemoveOrderID(oid order.OrderID) (*order.LimitOrder, bool) {
 	return pq.removeOrder(pq.orders[oid])
 }
 
-// HaveOrder indicates if an order is in the queue.
-func (pq *OrderPQ) HaveOrder(oid order.OrderID) bool {
+// RemoveUserOrders removes all orders from the queue that belong to a user.
+func (pq *OrderPQ) RemoveUserOrders(user account.AccountID) (removed []*order.LimitOrder) {
 	pq.mtx.Lock()
 	defer pq.mtx.Unlock()
-	return pq.orders[oid] != nil
+	uos, found := pq.userOrders[user]
+	if !found {
+		return
+	}
+	delete(pq.userOrders, user) // save some work from removeOrder=>Pop
```

Comment:
I don't follow.  Would you clarify?

Reply:
In `Pop`, at https://github.com/decred/dcrdex/blob/7eebbb067198c5c342f3fe14a65faaadc7ac7951/server/book/orderpq.go#L259, you are checking `found` and deleting if you find the user, but if you don't find the user, there is no indication. That is necessary, since you are pre-deleting the user here in `RemoveUserOrders`. But if you weren't doing that, and the user wasn't found, that would indicate some serious programmer error and would be something we'd want to know about and probably `panic`. For the `RemoveOrder` and `RemoveOrderID` paths that also use `Pop`, a missing user would be problematic. But we can't check it because of this pre-delete. Or am I reading this wrong?

Reply:
Gotcha.  Yeah, a `!found` here would indicate a bug if not for the possibility of removal in `RemoveUserOrders`.  It's not a major optimization for users with moderate order counts, so will remove this.  But despite not finding an entry in `userOrders` being an indication of programmer error, I'm hesitant to panic as that also implies there's no safe path forward, which their is.  To simply draw attention to programmer error, I'm inclined to loudly print to stdout.